pom.xml

```diff
@@ -470,6 +542,73 @@
                 </plugins>
             </build>
         </profile>
+        <!-- CI related -->
+        <profile>
+            <!-- This profile will run much more in-depth static analysis checks. -->
+            <id>static-analysis</id>
+            <properties>
+                <skipTests>true</skipTests>
+                <maven.javadoc.skip>true</maven.javadoc.skip>
+            </properties>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>com.github.spotbugs</groupId>
+                        <artifactId>spotbugs-maven-plugin</artifactId>
+                        <version>${spotbugs.plugin.version}</version>
+                        <configuration>
+                            <failOnError>false</failOnError>
+                            <!-- https://spotbugs.readthedocs.io/en/stable/effort.html -->
+                            <effort>min</effort>
```

Comment:
Now that it does not run by default, perhaps it does not need to be `min`?

Reply:
I can evaluate this once it's hooked into the target pipeline.